uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/API/Query/Select.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+use Civi\API\Api3SelectQuery;
+
+/**
+ * This class is just a wrapper around Civi\API\Api3SelectQuery.
+ *
+ * It introduces a new constructor which accepts the API params and then uses it
+ * to prepare the internal query properties.
+ */
+class CRM_HRLeaveAndAbsences_API_Query_Select extends Api3SelectQuery {
+
+  /**
+   * CRM_HRLeaveAndAbsences_API_Query_Select constructor.
+   *
+   * @param string $entity
+   *   An entity name like Contact, LeaveRequest, etc
+   * @param array $params
+   *   The API params array
+   */
+  public function __construct($entity, $params) {
+    $checkPermissions = !empty($params['check_permissions']);
+    parent::__construct($entity, $checkPermissions);
+    $this->setupQueryFromParams($params);
+  }
+
+  /**
+   * Setup things like limit, order, offset and fields to return
+   *
+   * @param array $params
+   */
+  private function setupQueryFromParams($params) {
+    $this->where = $params;
+
+    $options = _civicrm_api3_get_options_from_params($params);
+    $this->select = array_keys(array_filter($options['return']));
```

Comment:
Wondering if we need to make provision for 'is_count' option like its done here: [Here](https://github.com/civicrm/civicrm-core/pull/7725/files#diff-3076917760efeb467459726ebb6af282R1344)
?